h2o-algos/src/main/java/hex/tree/SharedTree.java

```diff
@@ -974,7 +974,8 @@ public static TwoDimTable createScoringHistoryTable(Model.Output _output,
 
     int rows = 0;
     for( int i = 0; i<_scored_train.length; i++ ) {
-      if (i != 0 && Double.isNaN(_scored_train[i]._rmse) && (_scored_valid == null || Double.isNaN(_scored_valid[i]._rmse))) continue;
+      if (i != 0 && Double.isNaN(_scored_train[i]._rmse) && Double.isNaN(_scored_train[i]._custom_metric) && 
+              (_scored_valid == null || (Double.isNaN(_scored_valid[i]._rmse) && Double.isNaN(_scored_valid[i]._custom_metric)))) continue;
       rows++;
```

Comment:
Would it make sense to wrap the complicated (and duplicated) conditional logic into a helper method that would provide a clue to a casual reader what's going on? E.g.
`if (i != 0 && isScoreMissingAtRow(i)) continue;`

Reply:
yes, you are right - let me see if I can improve it

Reply:
I will just use isEmpty from ScoreKeeper - in this context it should have the same meaning effectively